vultr/provider_test.go

```diff
@@ -1,16 +1,26 @@
 package vultr
 
 import (
+	"os"
 	"testing"
 
 	"github.com/hashicorp/terraform/helper/schema"
 	"github.com/hashicorp/terraform/terraform"
 )
 
-var _ terraform.ResourceProvider = Provider()
+var testAccProvider = Provider().(*schema.Provider)
+var testAccProviders = map[string]terraform.ResourceProvider{
+	"vultr": testAccProvider,
+}
 
 func TestProvider(t *testing.T) {
 	if err := Provider().(*schema.Provider).InternalValidate(); err != nil {
 		t.Fatalf("expected provider to validate: %v", err)
 	}
 }
+
+func testAccPreCheck(t *testing.T) {
```

Comment:
Right now, this function will never be executed :/ by default, the golang test tooling will run all exported functions beginning with `Test`, `test...` is not exported and is not run by go. We can make an `init` func that call this function, or put this precheck directly in the init func.

Reply:
It looks like it's common to define testAccPreCheck in provider_test.go: https://github.com/terraform-providers/terraform-provider-gitlab/blob/master/gitlab/provider_test.go
https://github.com/terraform-providers/terraform-provider-datadog/blob/master/datadog/provider_test.go

Then we would use it in the acceptance tests:
```
func TestAccExampleWidget_basic(t *testing.T) {
  resource.Test(t, resource.TestCase{
    PreCheck:     func() { testAccPreCheck(t) },
    // ...
  }
}
```

I think since it's only being used within the package, it doesn't need to be exported.

Please take a moment to review the links above when you get a chance. I think I've seen this pattern in every official provider I've reviewed. If you still think I should move it to an init() function, let me know and I will update this PR.

Reply:
Hi @afady, in the example you posted, the function you defined is actually invoked. I am not suggesting that we export the function, I'm just saying that nothing is ever executing it as it stands today; it is dead code that is never exercised. Either we:
1. make it an init
3. call it from somewhere else
2. get rid of it for now

Makes sense?

Reply:
I understand. My intention with this PR is to get the boilerplate out of the way so we can begin writing acceptance tests (#27 would also need to be merged). Part of that is defining the testAccPreCheck func. I will remove it for now and re-add it when I have the first acceptance test ready.

Reply:
Perfect, let's put a pin in it and add that func back once we begin adding the tests. Thanks again @afady !